lib/vnet/app_resolver.go

```diff
@@ -282,9 +282,15 @@ func (r *TCPAppResolver) resolveTCPHandlerForCluster(
 }
 
 type tcpAppHandler struct {
-	profileName     string
-	leafClusterName string
-	lp              *alpnproxy.LocalProxy
+	log              *slog.Logger
+	appProvider      AppProvider
+	clock            clockwork.Clock
+	profileName      string
+	leafClusterName  string
+	app              types.Application
+	portToLocalProxy map[uint16]*alpnproxy.LocalProxy
+	// mu guards access to portToLocalProxy.
+	mu sync.Mutex
```

Comment:
Consider grouping `mu` with the fields it protects. 

```suggestion
	app              types.Application
	
	// mu guards access to the fields that follow
	mu sync.Mutex
	portToLocalProxy map[uint16]*alpnproxy.LocalProxy
```

Reply:
Oops, forgot to address this. But I think spelling it out explicitly in the comment might be slightly better – at least you can see which fields it protects when you do a lookup on the field in your editor.